pulpcore/cli/common/generic.py

```diff
@@ -863,11 +863,13 @@ def _type_callback(ctx: click.Context, param: click.Parameter, value: t.Optional
     help=_("Name of the repository"),
     callback=lookup_callback("name", PulpRepositoryContext),
     expose_value=False,
+    is_eager=True,
```

Comment:
I'm not convinced, this is a nice solution, because it put's `--repository` on the same level as `--help`.
I believe we have ways to specify repository versions in a single option with the export code.

Reply:
Would that mean that the help message will not be printed if I use both `--repository` and `--help`?

Reply:
That's sure something to test. But I'm hesitant to call "--repository" _that_ special.

Reply:
@mdellweg I though so based on the documentation, but --help still acts as it should. Even if I use --repository with an invalid repository name, and put --help AFTER --repository, I still only get the error message printed.

Maybe --help is not only eager, but also has some special property in Click.

Reply:
"repo" is a repository that actually exists, "repo2" is nonexistent
```
pulp file repository version show --repository repo --help
Usage: pulp file repository version show [OPTIONS]

  Show details of a repository version.

Options:
  --repository-href TEXT  HREF of the repository
  --repository TEXT       A resource to look for identified by <name> or by
                          <href>.
  --version INTEGER       Repository version number
  --help                  Show this message and exit.
  ```

```
pulp file repository version show --repository repo2 --help
Usage: pulp file repository version show [OPTIONS]

  Show details of a repository version.

Options:
  --repository-href TEXT  HREF of the repository
  --repository TEXT       A resource to look for identified by <name> or by
                          <href>.
  --version INTEGER       Repository version number
  --help                  Show this message and exit.
```

Reply:
OK, eager is not the solution, because the real issue in disguise is that `_version_callback` is evaluating the repository context prematurely (before _all_ options have been seen), by accessing it's `pulp_href`. Leading to api calls before we even know the command is syntactically correct.
Instead, it should hook the contexts up lazily.